src/js/router/router.js

```diff
@@ -46,6 +46,8 @@ export const handleLocation = async () => {
     urlParams.set('filter', 'Muscles');
     urlParams.set('limit', '10');
     urlParams.set('page', '1');
+    window.location.search = urlParams.toString();
+    window.history.pushState({}, '', window.location);
```

Comment:
Please remove it from here, history push goes to route function. Add params to data set, create array with string of query param title, ie
`const dataParams = ['bodypart', 'filter', 'page'] `
then iterrate this array trough data set object and add it to url search params using `append` method
In route function instead of
`const { pathname } = new URL(href);
  console.log(pathname);
  window.history.pushState({}, '', `${basePath}${pathname}`);`
  
  `const url = new URL(href);`
iterrate data set with params and append them to url.searchParams
  `window.history.pushState({}, '', url);`